specs/specs.js

```diff
@@ -73,83 +83,106 @@ describe('Fingerprint2', function () {
       })
 
       it('does not try calling flash font detection', function (done) {
+        var flashFontKey = 'fontsFlash'
         var fp2 = new Fingerprint2()
-        spyOn(fp2, 'flashFontsKey')
-        fp2.get(function (result) {
-          expect(fp2.flashFontsKey).not.toHaveBeenCalled()
+        fp2.get(function (result, components) {
+          expect(components.some(function (componentResult) {
+            return componentResult.key === flashFontKey
+          })).toBeFalse()
           done()
         })
       })
     })
 
     describe('non-default options', function () {
       it('does not use userAgent when excluded', function (done) {
-        var fp2 = new Fingerprint2({excludeUserAgent: true})
-        spyOn(fp2, 'getUserAgent')
-        fp2.get(function (result) {
-          expect(fp2.getUserAgent).not.toHaveBeenCalled()
+        var userAgentKey = 'userAgent'
+        var fp2 = new Fingerprint2({excludes: [userAgentKey]})
+        fp2.get(function (result, components) {
+          expect(components.some(function (componentResult) {
+            return componentResult.key === userAgentKey
+          })).toBeFalse()
           done()
         })
       })
 
       it('does not use pixelRatio when excluded', function (done) {
-        var fp2 = new Fingerprint2({excludePixelRatio: true})
-        spyOn(fp2, 'getPixelRatio')
-        fp2.get(function (result) {
-          expect(fp2.getPixelRatio).not.toHaveBeenCalled()
+        var key = 'pixelRatio'
```

Comment:
I guess we can remove all of the duplicated exclusion tests.

Reply:
Or maybe make it a loop based single test case instead. Repeating all the test code for each and any of the exclusion options is useless.